--- a/IGC/BiFModule/CMakeLists.txt
+++ b/IGC/BiFModule/CMakeLists.txt
@@ -24,13 +24,13 @@
     # Get common clang library soname
     get_target_property(CCLANG_LIB_PATH opencl-clang-lib "IMPORTED_LOCATION")
     execute_process(
-      COMMAND readelf -d ${CCLANG_LIB_PATH}
+      COMMAND bash -c "readelf -d ${CCLANG_LIB_PATH} | grep SONAME"
       RESULT_VARIABLE CCLANG_READELF_RESULT
       OUTPUT_VARIABLE CCLANG_READELF_CALL)
     if(CCLANG_READELF_RESULT AND NOT CCLANG_READELF_RESULT EQUAL 0)
       message(FATAL_ERROR "[IGC\\BiFModule] : Error occurred while executing readelf:  ${CCLANG_READELF_RESULT}")
     endif()
-    string(REGEX MATCH "\\[${COMMON_CLANG_LIB_FULL_NAME}\\.([0-9](\\.[0-9]*)*[a-zA-Z0-9]*)\\]" CCLANG_SONAME_VERSION "${CCLANG_READELF_CALL}")
+    string(REGEX MATCH "\\[${COMMON_CLANG_LIB_FULL_NAME}\\.([0-9][0-9]\\.[0-9])\\]" CCLANG_SONAME_VERSION "${CCLANG_READELF_CALL}")
     set(CCLANG_SONAME_VERSION "${CMAKE_MATCH_1}")

     # Check if common clang library is newer than 5.0.0 version on which we have SPIR-V support
